luda-editor/luda-editor-client/src/app/sequence_list/ops/sync_sequences_from_google_spreadsheet.rs

```diff
@@ -53,37 +54,35 @@ impl SequenceList {
 }
 
 async fn sync_sequences_with_sheets(
-    socket: &Socket,
+    storage: &Arc<dyn GithubStorage>,
 ) -> Result<LinkedHashMap<String, Arc<Sequence>>, String> {
     let sheets = google_spreadsheet::get_sheets().await?;
 
-    save_order_of_spreadsheet_to_local(&socket, &sheets).await?;
+    save_order_of_spreadsheet_to_local(&storage, &sheets).await?;
 
-    let mut title_sequence_map = get_sequences_with_title(&socket).await?;
+    let mut title_sequence_map = get_sequences_with_title(&storage).await?;
 
     create_new_sequences_if_not_exist_in_sequences_but_in_sheets(&mut title_sequence_map, &sheets);
 
     delete_sequence_if_not_exist_in_sheets(&mut title_sequence_map, &sheets);
 
     sync_sequences_subtitles_from_sheets(&sheets, &mut title_sequence_map)?;
 
-    save_sequences(&socket, &title_sequence_map).await?;
+    save_sequences(&storage, &title_sequence_map).await?;
 
     Ok(title_sequence_map)
 }
 
 async fn save_sequences(
-    socket: &Socket,
+    storage: &Arc<dyn GithubStorage>,
     title_sequence_map: &LinkedHashMap<String, Arc<Sequence>>,
 ) -> Result<(), String> {
-    socket
-        .put_sequences(luda_editor_rpc::put_sequences::Request {
-            title_sequence_json_tuples: title_sequence_map
-                .iter()
-                .map(|(title, sequence)| (title.clone(), sequence.into_json()))
-                .collect(),
-        })
-        .await?;
+    for (title, sequence) in title_sequence_map {
+        storage
+            .put_sequence(title.as_str(), sequence)
+            .await
+            .map_err(|error| format!("failed to save sequence: {:#?}", error))?;
+    }
     Ok(())
 }
```

Comment:
save sequences one by one, not in parallel.
Conflict may occur when a write api is requested at the same time.
However, we don't have strategy to respond the conflict.
So we save the sequences  one by one.